File: magmac/src/java/com/meti/StateBuilder.java
package com.meti;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class StateBuilder {
    private String value = "";
    private Optional<String> exports = Optional.empty();
    private List<String> functions = new ArrayList<>();
    private Optional<String> importString = Optional.empty();
    private Optional<String> structs = Optional.empty();

    public StateBuilder setValue(String value) {
        this.value = value;
        return this;
    }

    public StateBuilder setExports(Optional<String> exports) {
        this.exports = exports;
        return this;
    }

    public StateBuilder setFunctions(List<String> functions) {
        this.functions = functions;
        return this;
    }

    public StateBuilder setImportString(Optional<String> importString) {
        this.importString = importString;
        return this;
    }

    public StateBuilder setStructs(Optional<String> structs) {
        this.structs = structs;
        return this;
    }

    public ApplicationState createState() {
        return new ApplicationState(importString, value, exports, structs, functions);
    }
}

File: magmac/src/c/com/meti/StateBuilder.c
#include "StateBuilder.c"
StateBuilder setExports(struct StateBuilder* __self__, Optional<String> exports){
        this.exports = exports;
        return this;
    }
StateBuilder setFunctions(struct StateBuilder* __self__, List<String> functions){
        this.functions = functions;
        return this;
    }
StateBuilder setImportString(struct StateBuilder* __self__, Optional<String> importString){
        this.importString = importString;
        return this;
    }
StateBuilder setStructs(struct StateBuilder* __self__, Optional<String> structs){
        this.structs = structs;
        return this;
    }
ApplicationState createState(struct StateBuilder* __self__){
        return new ApplicationState(importString, value, exports, structs, functions);
    }
struct StateBuilder StateBuilder(){
    private String value = ""
    private Optional<String> exports = Optional.empty()
    private List<String> functions = new ArrayList<>()
    private Optional<String> importString = Optional.empty()
    private Optional<String> structs = Optional.empty()
	pub def setValue(value : String) : StateBuilder => {
        this.value = value;
        return this;
    }
	struct StateBuilder this = {};
	return this;
}